### DISCUSSION TITLE
osu! fails to start on Arch Linux

### DISCUSSION THREAD

**Discussion by caitsa** [ppy/osu#27689] [Q&A]
### Type

Crash to desktop

### Bug description

Both the AppImage provided here and the AUR versions of osu! fail to start on Arch, without any log information in the logs folder or when run in a terminal. Removing my osu! directory in ~/.local/share doesn't fix this problem.

### Version

Every version from '2024.130.2' to '2024.312.1', currently the latest

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14703781/compressed-logs.zip)
These are from older versions of osu!. The versions that fail to start don't leave logs.

**Comment by bdach** [comment]
If there are no logs we have nothing to work with.

Please check `dmesg`, `journalctl`, or other system-level logging of your distro's choice.

**Reply by caitsa** [reply]
dmesg doesn't seem to give anything pertaining to osu. journalctl gives the following two lines:
```
systemd[923]: Started /home/hylic/games/osu.AppImage.
systemd[1]: tmp-.mount_osu.ApOiVCH8.mount: Deactivated successfully.
```

**Comment by sineplusx** [comment]
Are you sure there is no error information when you run the AppImage from a terminal? Please try executing `echo $?` in the same terminal after you have run the AppImage from there.

**Reply by caitsa** [reply]
It says '0'.

**Reply by sineplusx** [reply]
Sorry for doubting your initial description. It really looks like there is no error being reported.

You mentioned that the last working release is 2024.114.0. Every new version since then is distributed with anticheat included, and I guess the issue is related to that. However, when something goes wrong with it, its usually visible in logs.

Here are a few things you can try:

1. After launching the game regularly, do you have a file named `AuthNative.so` in `~/.local/share/osu/` ? If yes, there should also be a hidden file `.auth_startup` that would probably be of interest.
2. Delete the `osu` folder in `~/.local/share/`, then try opening the game while disconnected from the internet (Backup everything if you want to keep beatmaps, scores, etc. obviously)
3. Any Linux security modules installed and active? For example AppArmor or SELinux

**Reply by caitsa** [reply]
There's no anticheat files in the osu folder, and step 2 doesn't change anything. I don't think I have any non-default modules installed except for lkrg, and osu didn't work before I installed it. I tried step 2 again after unloading lkrg, and no dice.

**Reply by sineplusx** [reply]
> I don't think I have any non-default modules installed except for lkrg, and osu didn't work before I installed it

So the issue was already present _before_ you installed LKRG right?

Im kinda running out of ideas here. To make sure it really is the anticheat acting up, you can try the following:

- Download the [source code](https://github.com/ppy/osu/archive/refs/tags/2024.312.1.zip) of the latest official version and install the `dotnet-sdk` package
- Compile the source code into a local release by executing `dotnet publish -c Release --sc true -o ./build -r linux-x64` in the `osu.Desktop` directory
- Again backup and clean `~/.local/share/osu`
- Navigate to the generated `build` directory and try executing the `osu!` binary.

If this works and it starts correctly, it will be a problem with the anticheat that we have to narrow down further.
If it doesnt start, I honestly have no clue.

**Reply by caitsa** [reply]
It doesn't run, and it's just as unresponsive as always. Thanks for trying to help.

**Reply by Rekunan** [reply]
Sounds like it's an issue with the Linux install, and at that point, your best bet would be to debug it yourself.

**Reply by caitsa** [reply]
I neglected to mention this earlier, but the development build of the game runs for me.

> your best bet would be to debug it yourself.

How?

**Reply by Rekunan** [reply]
> How?

Usually by looking through your config files, installed packages, etc., making sure there isn't anything messing with them or whatever.

However, since the development build works, it means either the performance optimizations or the game files at fault. I'm leaning towards the latter, so I'll ask that you make sure that the game files are fine (no corruptions or whatever) with nothing messing with them (antivirus, other programs, etc.), and making sure they have the right permissions.

@sineplusx pinging in case you have anything else to add.

**Reply by sineplusx** [reply]
Guides for debugging can be found on the [ArchWiki](https://wiki.archlinux.org/title/Debugging) and various other sites. I would try running osu! with the tools mentioned there to get a hint what causes the exit. Aside from that, you can use search engines and other forums. Maybe try some other .NET applications and see if the same issue crops up outside of osu!.

One more tip, if you want to test stuff with the official release, you can get a better testing environment by extracting all game files: `./osu.AppImage --appimage-extract`
With the .NET SDK you can also run the game like this from the extracted files: `dotnet ./osu!.dll` (runs it without the included loader binary)

**Comment by smoogipoo** [comment]
The only other suggestion I have is, if you're using Wayland, to try starting the game with the environment variable `SDL_VIDEODRIVER=wayland`.

Otherwise, unfortunately it looks like you're on your own with this one.

**Comment by caitsa** [comment]
Running the AppImage with "LD_DEBUG=files" set gives the following errors:
```
osu!: error: symbol lookup error: undefined symbol: DllMain (fatal)
/tmp/.mount_osu.ApTA7mbI/usr/bin/libcoreclr.so: error: symbol lookup error: undefined symbol: PAL_RegisterModule (fatal)
/tmp/.mount_osu.ApTA7mbI/usr/bin/libclrjit.so: error: symbol lookup error: undefined symbol: DllMain (fatal)
/usr/lib/libicuuc.so.75: error: symbol lookup error: undefined symbol: u_strlen (fatal)
```
If anyone else is having this issue, either try running the Windows version through Wine, build the debug version of the game from source (which works for me), or install Fedora.

**Comment by caitsa** [comment]
AppImage for version 2025.221.0 runs.
